src/Features/Core/Portable/CodeRefactorings/SyncNamespace/AbstractSyncNamespaceService.ChangeNamespaceCodeAction.cs

```diff
@@ -0,0 +1,526 @@
+                    var refNode = root.FindNode(refLoc.Location.SourceSpan, getInnermostNodeForTie: true);
+                    if (syncNamespaceService.TryGetReplacementReferenceSyntax(
+                        refNode, namespaceParts, syntaxFacts, out var oldNode, out var newNode))
+                    {
+                        editor.ReplaceNode(oldNode, newNode.WithAdditionalAnnotations(Simplifier.Annotation));
+                    }
+
+                    // Use a dummy import node to figure out which container the new import will be added to.
+                    var container = addImportService.GetImportContainer(root, refNode, dummyImport);
+                    containers.Add(container);
+                }
+
+                foreach(var container in containers)
+                {
+                    editor.TrackNode(container);
+                }
+
+                document = editor.GetChangedDocument();
+                root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+                return (document, containers.Select(c => root.GetCurrentNode(c)).ToImmutableArray());
+            }
+
+            private async Task<Solution> RemoveUnnecessaryImportsAsync(
+                Solution solution, 
+                ImmutableArray<DocumentId> ids, 
+                ImmutableArray<string> names, 
+                CancellationToken cancellationToken)
+            {
+                var LinkedDocumentsToSkip = PooledHashSet<DocumentId>.GetInstance();
+                foreach (var id in ids)
+                {
+                    if (LinkedDocumentsToSkip.Contains(id))
+                    {
+                        continue;
+                    }
+
+                    var document = solution.GetDocument(id);
+                    LinkedDocumentsToSkip.AddRange(document.GetLinkedDocumentIds());
+
+                    document = await RemoveUnnecessaryImportsAsync(
+                        document, 
+                        CreateImports(document, names, withFormatterAnnotation: false), 
+                        cancellationToken)
+                        .ConfigureAwait(false);
+                    solution = document.Project.Solution;
+                }
+
+                LinkedDocumentsToSkip.Free();
+                return solution;
+            }
+
+            private async Task<Document> RemoveUnnecessaryImportsAsync(
+                Document document, 
+                IEnumerable<SyntaxNode> importsToRemove,
+                CancellationToken cancellationToken)
+            {
+                var removeImportService = document.GetLanguageService<IRemoveUnnecessaryImportsService>();
+                return await removeImportService.RemoveUnnecessaryImportsAsync(
+                    document,
+                    import => importsToRemove.Any(importToRemove => importToRemove.IsEquivalentTo(import, topLevel: false)),
+                    cancellationToken)
+                    .ConfigureAwait(false);
```

Comment:
wrapping args i get.  i'm not sure why chained method calls are wrapped when there's enough room.